
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/IndigiDataAotearoa_IntroToR/3_AnalysingTwoDataTypes/">
      
      <link rel="icon" href="../theme_figures/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.3.1, mkdocs-material-8.3.9">
    
    
      
        <title>R for statistical tools and tests - IndigiDataAotearoa_IntroToR</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.1d29e8d0.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
      
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="custom" data-md-color-accent="">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#r-for-statistical-tools-and-tests" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
<!-- Custom Banner -->
<div class="banner">
    <p style="color:rgb(250, 246, 246); display:inline;font-size: 160%;">This repository is in the early stages of development (Alpha Version)</p>
</div>

<!-- Original Header -->


<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="IndigiDataAotearoa_IntroToR" class="md-header__button md-logo" aria-label="IndigiDataAotearoa_IntroToR" data-md-component="logo">
      
  <img src="../theme_figures/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            IndigiDataAotearoa_IntroToR
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              R for statistical tools and tests
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="" data-md-color-primary="custom" data-md-color-accent=""  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_3" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent=""  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent=""  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_3">
          
            <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/IndigiDataAotearoa_IntroToR" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/IndigiDataAotearoa_IntroToR
  </div>
</a>
      </div>
    
  </nav>
  
</header>

    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="IndigiDataAotearoa_IntroToR" class="md-nav__button md-logo" aria-label="IndigiDataAotearoa_IntroToR" data-md-component="logo">
      
  <img src="../theme_figures/nesi_ga.png" alt="logo">

    </a>
    IndigiDataAotearoa_IntroToR
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/IndigiDataAotearoa_IntroToR" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/IndigiDataAotearoa_IntroToR
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../1_Background/" class="md-nav__link">
        Introduction to the R programming languange
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../2_WorkingWithData/" class="md-nav__link">
        Analysing genetic variant data in R
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          R for statistical tools and tests
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        R for statistical tools and tests
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#population-diversity" class="md-nav__link">
    Population diversity
  </a>
  
    <nav class="md-nav" aria-label="Population diversity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#principal-components-analysis-pca" class="md-nav__link">
    Principal components analysis (PCA)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#phylogenetic-trees" class="md-nav__link">
    Phylogenetic trees
  </a>
  
    <nav class="md-nav" aria-label="Phylogenetic trees">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#key-takeaways" class="md-nav__link">
    Key takeaways
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#population-diversity" class="md-nav__link">
    Population diversity
  </a>
  
    <nav class="md-nav" aria-label="Population diversity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#principal-components-analysis-pca" class="md-nav__link">
    Principal components analysis (PCA)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#phylogenetic-trees" class="md-nav__link">
    Phylogenetic trees
  </a>
  
    <nav class="md-nav" aria-label="Phylogenetic trees">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#key-takeaways" class="md-nav__link">
    Key takeaways
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
  <a href="https://github.com/GenomicsAotearoa/IndigiDataAotearoa_IntroToR/edit/master/docs/3_AnalysingTwoDataTypes.md" title="Edit this page" class="md-content__button md-icon">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>



<h1 id="r-for-statistical-tools-and-tests">R for statistical tools and tests<a class="headerlink" href="#r-for-statistical-tools-and-tests" title="Permanent link">&para;</a></h1>
<div class="admonition info">
<p class="admonition-title">Info</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Keypoints</label><label for="__tabbed_1_2">Objectives</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<ul>
<li>R is popular for statistical tests</li>
<li>PCA can make your data easier to comprehend</li>
<li>Genetic variation can be population-specific</li>
</ul>
</div>
<div class="tabbed-block">
<ul>
<li>Investigate diversity in population-specific genetic variation</li>
<li>Understand PCA and how to run it in R</li>
</ul>
</div>
</div>
</div>
</div>
<h2 id="population-diversity">Population diversity<a class="headerlink" href="#population-diversity" title="Permanent link">&para;</a></h2>
<p>We will now load a second dataset which contains a new set of SNPs. Rather than being medically relevant, these SNPs were chosen because they are informative about ancestry. That is, variation in their genotype frequencies tends to be associated with differences between population groups. </p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1">## Load ancestry data</span>
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="n">snpAns</span> <span class="o">=</span> <span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;GENE360snpAncestry.txt&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="c1">## Check dimensionality of data</span>
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a><span class="nf">dim</span><span class="p">(</span><span class="n">snpAns</span><span class="p">)</span>
</code></pre></div>
</div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>[1] 2504 2305
</code></pre></div>
</details>
<p>Now we now have a <em>much</em> larger dataset: 2504 individuals and 2302 SNP genotypes for each individual. 
<strong>Note</strong>: I've said there are 2302 SNP genotypes, but the dim function above returned 2305 columns. Think back to some of the functions you have used earlier and use one (or more) to check what the other three columns are. </p>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code>A)

!!! r-project

    ```r
    View(snpAns)
    ```

B)

!!! r-project

    ```r
    names(snpAns)
    ```
</code></pre></div>
</details>
<p>This new dataset includes not just the super populations we worked with previously, but also sub-populations. We can create a table that breaks down each super population into subpopulations:</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="nf">table</span><span class="p">(</span><span class="n">snpAns</span><span class="o">$</span><span class="n">SubPopulation</span><span class="p">,</span> <span class="n">snpAns</span><span class="o">$</span><span class="n">Population</span><span class="p">)</span>
</code></pre></div>
</div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>##          
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>##           AFR AMR EAS EUR SAS
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>##   AFR_ACB  96   0   0   0   0
<a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a>##   AFR_ASW  61   0   0   0   0
<a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>##   AFR_ESN  99   0   0   0   0
<a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>##   AFR_GWD 113   0   0   0   0
<a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a>##   AFR_LWK  99   0   0   0   0
<a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a>##   AFR_MSL  85   0   0   0   0
<a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a>##   AFR_YRI 108   0   0   0   0
<a id="__codelineno-5-10" name="__codelineno-5-10" href="#__codelineno-5-10"></a>##   AMR_CLM   0  94   0   0   0
<a id="__codelineno-5-11" name="__codelineno-5-11" href="#__codelineno-5-11"></a>##   AMR_MXL   0  64   0   0   0
<a id="__codelineno-5-12" name="__codelineno-5-12" href="#__codelineno-5-12"></a>##   AMR_PEL   0  85   0   0   0
<a id="__codelineno-5-13" name="__codelineno-5-13" href="#__codelineno-5-13"></a>##   AMR_PUR   0 104   0   0   0
<a id="__codelineno-5-14" name="__codelineno-5-14" href="#__codelineno-5-14"></a>##   EAS_CDX   0   0  93   0   0
<a id="__codelineno-5-15" name="__codelineno-5-15" href="#__codelineno-5-15"></a>##   EAS_CHB   0   0 103   0   0
<a id="__codelineno-5-16" name="__codelineno-5-16" href="#__codelineno-5-16"></a>##   EAS_CHS   0   0 105   0   0
<a id="__codelineno-5-17" name="__codelineno-5-17" href="#__codelineno-5-17"></a>##   EAS_JPT   0   0 104   0   0
<a id="__codelineno-5-18" name="__codelineno-5-18" href="#__codelineno-5-18"></a>##   EAS_KHV   0   0  99   0   0
<a id="__codelineno-5-19" name="__codelineno-5-19" href="#__codelineno-5-19"></a>##   EUR_CEU   0   0   0  99   0
<a id="__codelineno-5-20" name="__codelineno-5-20" href="#__codelineno-5-20"></a>##   EUR_FIN   0   0   0  99   0
<a id="__codelineno-5-21" name="__codelineno-5-21" href="#__codelineno-5-21"></a>##   EUR_GBR   0   0   0  91   0
<a id="__codelineno-5-22" name="__codelineno-5-22" href="#__codelineno-5-22"></a>##   EUR_IBS   0   0   0 107   0
<a id="__codelineno-5-23" name="__codelineno-5-23" href="#__codelineno-5-23"></a>##   EUR_TSI   0   0   0 107   0
<a id="__codelineno-5-24" name="__codelineno-5-24" href="#__codelineno-5-24"></a>##   SAS_BEB   0   0   0   0  86
<a id="__codelineno-5-25" name="__codelineno-5-25" href="#__codelineno-5-25"></a>##   SAS_GIH   0   0   0   0 103
<a id="__codelineno-5-26" name="__codelineno-5-26" href="#__codelineno-5-26"></a>##   SAS_ITU   0   0   0   0 102
<a id="__codelineno-5-27" name="__codelineno-5-27" href="#__codelineno-5-27"></a>##   SAS_PJL   0   0   0   0  96
<a id="__codelineno-5-28" name="__codelineno-5-28" href="#__codelineno-5-28"></a>##   SAS_STU   0   0   0   0 102
</code></pre></div>
</details>
<p>To examine population diversity we need to do two things:</p>
<ol>
<li>
<p>Create a data object of <em>only</em> the SNP genotype data (i.e., remove the first three columns).</p>
</li>
<li>
<p>Convert the genotypes to allele counts (e.g., if we wanted to count the number of A's: TT, AT, AA becomes 0, 1, 2, respectively).</p>
</li>
</ol>
<p>Step 1:</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="c1">## Create a new object called snpAnsDat which contains only the SNP data. </span>
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a><span class="c1">## To do this we will take all the data from snpAns except rows 1, 2 and 3:</span>
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a><span class="n">snpAnsData</span> <span class="o">=</span> <span class="n">snpAns</span><span class="p">[,</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">3</span><span class="p">)]</span>
</code></pre></div>
</div>
<p>Step 2:</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1">## Load a custom function to convert genotypes (AA, AT, TT) into allele counts (2, 1, 0 if we are counting A&#39;s).</span>
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="nf">source</span><span class="p">(</span><span class="s">&#39;alleleCounts.R&#39;</span><span class="p">)</span>
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a><span class="c1">## Apply the function to the genotype data, one column (SNP) at a time. </span>
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a><span class="n">snpAnsCount</span> <span class="o">=</span> <span class="nf">apply</span><span class="p">(</span><span class="n">snpAnsData</span><span class="p">,</span> <span class="m">2</span><span class="p">,</span> <span class="n">alleleCounts</span><span class="p">)</span>
</code></pre></div>
</div>
<p>We now have a new object that contains just the SNP data represented by allele counts. </p>
<p>Let's look at this new data set using View():</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="nf">View</span><span class="p">(</span><span class="n">snpAnsCount</span><span class="p">)</span>
</code></pre></div>
</div>
<h3 id="principal-components-analysis-pca">Principal components analysis (PCA)<a class="headerlink" href="#principal-components-analysis-pca" title="Permanent link">&para;</a></h3>
<p>In terms of examining population diversity, we have 2302 dimensions of
data available - one dimension for each SNP.</p>
<p>Rather than trying to comprehend this huge amount of data in
2302-dimensional space, genetics researchers often use a statistical tool called Principal Components Analysis (PCA) to reduce the dimensionality of the data.</p>
<p>The idea is to find the most important variation in the data, and examine the samples in terms of that variation, ignoring the rest. In practice, this works fairly well, because genetic differences between populations provide a strong (and relatively consistent) source of
variation across genomic loci (i.e., SNPs). Rather than looking at 2302 dimensions of data, we end up looking at variation across just 2 or 3 dimensions - each dimension is defined by a combination of SNPs which vary in a similar way across the individuals in the study.</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="c1">## Load custom function to perform principal components analysis</span>
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a><span class="nf">source</span><span class="p">(</span><span class="s">&quot;pcaGenotypes.R&quot;</span><span class="p">)</span>
<a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>
<a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a><span class="c1">## Apply this function to the allele count data</span>
<a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a><span class="c1">## This will take a couple of minutes to run</span>
<a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a><span class="n">pca</span> <span class="o">=</span> <span class="nf">pcaGenotypes</span><span class="p">(</span><span class="n">snpAnsCount</span><span class="p">)</span>
</code></pre></div>
</div>
<p>To visualise the populations across the principal components, we need to define colours for each populations. It doesn't matter what colours you choose, but lets go with: AFR (brown), AMR (red), EAS (purple), EUR (blue), SAS (green).</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
</div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1">## Create an object relating to the population data</span>
<a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="n">ansPop</span> <span class="o">=</span> <span class="n">snpAns</span><span class="o">$</span><span class="n">Population</span>
<a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>
<a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a><span class="c1">## Generate colours to associate with each population</span>
<a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a><span class="n">pCols</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;brown&quot;</span><span class="p">,</span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="s">&quot;purple&quot;</span><span class="p">,</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="s">&quot;green&quot;</span><span class="p">)</span>
<a id="__codelineno-10-6" name="__codelineno-10-6" href="#__codelineno-10-6"></a><span class="n">popCol</span> <span class="o">=</span> <span class="n">pCols</span><span class="p">[</span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">as.factor</span><span class="p">(</span><span class="n">ansPop</span><span class="p">))]</span>
<a id="__codelineno-10-7" name="__codelineno-10-7" href="#__codelineno-10-7"></a><span class="nf">names</span><span class="p">(</span><span class="n">popCol</span><span class="p">)</span> <span class="o">=</span> <span class="n">ansPop</span>
<a id="__codelineno-10-8" name="__codelineno-10-8" href="#__codelineno-10-8"></a>
<a id="__codelineno-10-9" name="__codelineno-10-9" href="#__codelineno-10-9"></a><span class="c1">## Check that they correspond to populations</span>
<a id="__codelineno-10-10" name="__codelineno-10-10" href="#__codelineno-10-10"></a><span class="nf">table</span><span class="p">(</span><span class="n">popCol</span><span class="p">,</span><span class="n">ansPop</span><span class="p">)</span>
</code></pre></div>
<details class="success">
<summary>Output</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>##         ansPop
<a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>## popCol   AFR AMR EAS EUR SAS
<a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>##   blue     0   0   0 503   0
<a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>##   brown  661   0   0   0   0
<a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>##   green    0   0   0   0 489
<a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a>##   purple   0   0 504   0   0
<a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a>##   red      0 347   0   0   0
</code></pre></div>
</details>
<p>Now we can plot the principal components and colour the points based on the population each sample belongs to.</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
</div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="c1">## Load a custom plotting function to generate the plots:</span>
<a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a><span class="nf">source</span><span class="p">(</span><span class="s">&#39;plotPCA.R&#39;</span><span class="p">)</span>
<a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a>
<a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a><span class="c1">## Plot the PCA data that we have generated, along with the population information</span>
<a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a><span class="nf">plotPCA</span><span class="p">(</span><span class="n">pca</span><span class="p">,</span> <span class="n">popCol</span><span class="p">)</span>
</code></pre></div>
<p><img alt="images" src="../figures/pcaPlot.png" /></p>
<p>From the plots we can see that samples from the same population tend to cluster together, and that the first three principal components do a reasonable job of capturing the genetic diversity between the populations.</p>
<p>With the <code>scatterplot3d</code> package, you can plot the first three principal components at once (i.e., combining the information from the three scatterplots above). This shows that the European (EUR), East Asian (EAS) and South Asian (SAS) super-populations are relatively homogeneous, while the Ad-Mixed American (AMR) and African (AFR) super-populations exhibit greater variation, suggesting admixture within these groups.</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
</div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">scatterplot3d</span><span class="p">)</span>
<a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a><span class="nf">scatterplot3d</span><span class="p">(</span><span class="n">pca</span><span class="p">[,</span><span class="m">1</span><span class="p">],</span> <span class="n">pca</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span> <span class="n">pca</span><span class="p">[,</span><span class="m">3</span><span class="p">],</span> <span class="n">color</span><span class="o">=</span><span class="n">popCol</span><span class="p">,</span> <span class="n">pch</span><span class="o">=</span><span class="m">16</span><span class="p">,</span>
<a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a>              <span class="n">cex.symbols</span><span class="o">=</span><span class="m">0.5</span><span class="p">,</span> <span class="n">xlab</span><span class="o">=</span><span class="s">&quot;PC1&quot;</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;PC2&quot;</span><span class="p">,</span> <span class="n">zlab</span><span class="o">=</span><span class="s">&quot;PC3&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="images" src="../figures/scatterplot3d.png" /></p>
<h2 id="phylogenetic-trees">Phylogenetic trees<a class="headerlink" href="#phylogenetic-trees" title="Permanent link">&para;</a></h2>
<p>Another way to visualise genetic similarity is via “phylogenetic trees”, also known more generally as “dendrograms”.</p>
<p>These tree diagrams group items together based on similarity scores. In this setting our “items” are the populations, and the scores are calculated based on the genetic similarities between each pair of populations.</p>
<p>To calculate similarities, we need to create a set of “average” genotypes for each population. One way to do this is to calculate the frequency of the major homozygote for each SNP in each population.</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
</div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="c1">## Load custom function for calculating major homozygote frequency:</span>
<a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a><span class="nf">source</span><span class="p">(</span><span class="s">&#39;calcMajorFreq.R&#39;</span><span class="p">)</span>
<a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a>
<a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a><span class="c1">## Apply this function to the SNP count data</span>
<a id="__codelineno-14-5" name="__codelineno-14-5" href="#__codelineno-14-5"></a><span class="c1">## Takes a few seconds</span>
<a id="__codelineno-14-6" name="__codelineno-14-6" href="#__codelineno-14-6"></a><span class="n">popFreqs</span> <span class="o">=</span> <span class="nf">calcMajorFreq</span><span class="p">(</span><span class="n">snpAnsCount</span><span class="p">,</span> <span class="n">ansPop</span><span class="p">)</span>
</code></pre></div>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
</div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="c1">## Make a cluster tree based on these frequencies</span>
<a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="nf">plot</span><span class="p">(</span><span class="nf">hclust</span><span class="p">(</span><span class="nf">dist</span><span class="p">(</span><span class="n">popFreqs</span><span class="p">)),</span><span class="n">hang</span><span class="o">=</span><span class="m">-1</span><span class="p">)</span>
</code></pre></div>
<p><img alt="images" src="../figures/clusterDendrogram.png" /></p>
<p>The way in which the “leaves” of the tree cluster, reflects the similarity between the items analysed. In this case the populations are grouped based on their genetic similarity, as measured by these particular loci. Here it appears that the AMR (Ad-Mixed American), and SAS (South Asian) super-populations are most genetically similar, then
EUR (European), followed by EAS (East Asian). The AFR (African) super-population is the most genetically dissimilar relative to the others. It is likely that these groupings reflect (to some degree) migration patterns and shared ancestry experienced by these populations.</p>
<h3 id="key-takeaways">Key takeaways<a class="headerlink" href="#key-takeaways" title="Permanent link">&para;</a></h3>
<p>R has hundreds of functions that can do a wide range of tasks, from simple (view data, calculate mean, show you the range (min/max) values) to highly complex.</p>
<p>R is an excellent programming language for statistical analyses of large data sets.</p>
<p>R is <em>really</em> useful for making quick plots and tables to visualise your data.</p>
<p>Large, publicly available datasets are a tremendous resource. </p>

              
            </article>
            
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../2_WorkingWithData/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Analysing genetic variant data in R" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Analysing genetic variant data in R
            </div>
          </div>
        </a>
      
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      IndigiDataAotearoa_IntroToR workshop material is licensed under a <a rel="noopener" target="_blank" href="https://www.gnu.org/licenses/gpl-3.0.en.html">GNU General Public License v3.0</a>
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.footer", "content.code.copy", "content.code.annotate"], "search": "../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.6c7ad80a.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>