
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/IndigiDataAotearoa_IntroToR/1_Background/">
      
      <link rel="icon" href="../theme_figures/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.3.1, mkdocs-material-8.3.9">
    
    
      
        <title>Introduction to the R programming languange - Indigidata Aotearoa Intro To R</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.1d29e8d0.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
      
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="custom" data-md-color-accent="">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#introduction-to-the-r-programming-languange" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
<!-- Custom Banner -->
<div class="banner">
    <p style="color:rgb(250, 246, 246); display:inline;font-size: 160%;">This repository is in the early stages of development (Alpha Version)</p>
</div>

<!-- Original Header -->


<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Indigidata Aotearoa Intro To R" class="md-header__button md-logo" aria-label="Indigidata Aotearoa Intro To R" data-md-component="logo">
      
  <img src="../theme_figures/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Indigidata Aotearoa Intro To R
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Introduction to the R programming languange
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="" data-md-color-primary="custom" data-md-color-accent=""  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_3" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent=""  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent=""  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_3">
          
            <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/IndigiDataAotearoa_IntroToR" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/IndigiDataAotearoa_IntroToR
  </div>
</a>
      </div>
    
  </nav>
  
</header>

    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Indigidata Aotearoa Intro To R" class="md-nav__button md-logo" aria-label="Indigidata Aotearoa Intro To R" data-md-component="logo">
      
  <img src="../theme_figures/nesi_ga.png" alt="logo">

    </a>
    Indigidata Aotearoa Intro To R
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/IndigiDataAotearoa_IntroToR" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/IndigiDataAotearoa_IntroToR
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Introduction to the R programming languange
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Introduction to the R programming languange
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#a-brief-history-of-r" class="md-nav__link">
    A brief history of R
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#advantages-of-using-r" class="md-nav__link">
    Advantages of using R
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introducing-rstudio-server" class="md-nav__link">
    Introducing RStudio Server
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-an-rstudio-project" class="md-nav__link">
    Create an RStudio project
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-your-first-r-script" class="md-nav__link">
    Creating your first R script
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-to-work-with-r-navigating-directories" class="md-nav__link">
    Getting to work with R: navigating directories
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-functions-in-r-without-needing-to-master-them" class="md-nav__link">
    Using functions in R, without needing to master them
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-help-with-function-arguments" class="md-nav__link">
    Getting help with function arguments
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../2_WorkingWithData/" class="md-nav__link">
        Analysing genetic variant data in R
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../3_AnalysingTwoDataTypes/" class="md-nav__link">
        R for statistical tools and tests
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#a-brief-history-of-r" class="md-nav__link">
    A brief history of R
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#advantages-of-using-r" class="md-nav__link">
    Advantages of using R
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introducing-rstudio-server" class="md-nav__link">
    Introducing RStudio Server
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-an-rstudio-project" class="md-nav__link">
    Create an RStudio project
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-your-first-r-script" class="md-nav__link">
    Creating your first R script
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-to-work-with-r-navigating-directories" class="md-nav__link">
    Getting to work with R: navigating directories
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-functions-in-r-without-needing-to-master-them" class="md-nav__link">
    Using functions in R, without needing to master them
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-help-with-function-arguments" class="md-nav__link">
    Getting help with function arguments
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
  <a href="https://github.com/GenomicsAotearoa/IndigiDataAotearoa_IntroToR/edit/master/docs/1_Background.md" title="Edit this page" class="md-content__button md-icon">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>



<h1 id="introduction-to-the-r-programming-languange">Introduction to the R programming languange<a class="headerlink" href="#introduction-to-the-r-programming-languange" title="Permanent link">&para;</a></h1>
<div class="admonition info">
<p class="admonition-title">Info</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Keypoints</label><label for="__tabbed_1_2">Objectives</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<ul>
<li>R is a powerful, popular open-source scripting language</li>
<li>RStudio allows you to run R in an easy-to-use interface and makes it
  easy to find help</li>
<li>A popular language for it's use in statistical analysis and visualisation tools</li>
</ul>
</div>
<div class="tabbed-block">
<ul>
<li>Create an RStudio project, and know the benefits of working within a project</li>
<li>Be able to locate and change the current working directory with <code>getwd()</code> and <code>setwd()</code></li>
<li>Compose an R script file containing comments and commands</li>
<li>Understand what an R function is</li>
<li>Locate help for an R function using <code>?</code>, <code>??</code>, and <code>args()</code></li>
<li>Use R to visualise population-specific frequencies of medically relevant genetic variants</li>
</ul>
</div>
</div>
</div>
</div>
<h2 id="a-brief-history-of-r">A brief history of R<a class="headerlink" href="#a-brief-history-of-r" title="Permanent link">&para;</a></h2>
<p><a href="https://en.wikipedia.org/wiki/R_(programming_language)">R</a> has been
around since 1995, and was created by Ross Ihaka and Robert Gentleman at
the University of Auckland, New Zealand. R is based off the S
programming language developed at Bell Labs and was developed to teach
intro statistics.</p>
<h2 id="advantages-of-using-r">Advantages of using R<a class="headerlink" href="#advantages-of-using-r" title="Permanent link">&para;</a></h2>
<p>At more than 20 years old, R is fairly mature and <a href="https://www.tiobe.com/tiobe-index/r/">growing in
popularity</a>. However, programming
isn't a popularity contest. Here are key advantages of analyzing data in
R:</p>
<div class="admonition quote">
<ul>
<li><strong>R is <a href="https://en.wikipedia.org/wiki/Open-source_software">open
    source</a></strong>. This
    means R is free - an advantage if you are at an institution where
    you have to pay for your own MATLAB or SAS license. Open source is
    important to your colleagues in parts of the world where expensive
    software in inaccessible. It also means that R is actively developed
    by a community (see <a href="https://www.r-project.org/">r-project.org</a>),
    and there are regular updates.</li>
<li><strong>R is widely used</strong>. Ok, maybe programming is a popularity contest.
    Because R is used in many areas (not just bioinformatics), you are
    more likely to find help online when you need it. Chances are,
    almost any error message you run into, someone else has already
    experienced.</li>
<li><strong>R is powerful</strong>. R runs on multiple platforms
    (Windows/MacOS/Linux). It can work with much larger datasets than
    popular spreadsheet programs like Microsoft Excel, and because of
    its scripting capabilities is far more reproducible. Also, there are
    thousands of available software packages for science, including
genomics and other areas of life science.</li>
</ul>
</div>
<h2 id="introducing-rstudio-server">Introducing RStudio Server<a class="headerlink" href="#introducing-rstudio-server" title="Permanent link">&para;</a></h2>
<p>In these lessons, we will be making use of a software called
<a href="https://www.rstudio.com/products/RStudio/">RStudio</a>, an <a href="https://en.wikipedia.org/wiki/Integrated_development_environment">Integrated
Development Environment
(IDE)</a>.
RStudio, like most IDEs, provides a graphical interface to R, making it
more user-friendly, and providing dozens of useful features. We will
introduce additional benefits of using RStudio as you cover the lessons.</p>
<p><img alt="image" src="../figures/rstudio_session_default.png" width="700" /></p>
<div class="admonition quote">
<ul>
<li><strong>Source</strong>: This pane is where you will write/view R scripts. Some
    outputs (such as if you view a dataset using <code>View()</code>) will appear
    as a tab here.</li>
<li><strong>Console/Terminal/Jobs</strong>: This is actually where you see the
    execution of commands. This is the same display you would see if you
    were using R at the command line without RStudio. You can work
    interactively (i.e. enter R commands here), but for the most part we
    will run a script (or lines in a script) in the source pane and
    watch their execution and output here. The "Terminal" tab give you
    access to the BASH terminal (the Linux operating system, unrelated
    to R). RStudio also allows you to run jobs (analyses) in the
    background. This is useful if some analysis will take a while to
    run. You can see the status of those jobs in the background.</li>
<li><strong>Environment/History</strong>: Here, RStudio will show you what datasets
    and objects (variables) you have created and which are defined in
    memory. You can also see some properties of objects/datasets such as
    their type and dimensions. The "History" tab contains a history of
    the R commands you've executed R.</li>
<li><strong>Files/Plots/Packages/Help/Viewer</strong>: This multipurpose pane will
    show you the contents of directories on your computer. You can also
    use the "Files" tab to navigate and set the working directory. The
    "Plots" tab will show the output of any plots generated. In
    "Packages" you will see what packages are actively loaded, or you
    can attach installed packages. "Help" will display help files for R
    functions and packages. "Viewer" will allow you to view local web
    content (e.g. HTML outputs).</li>
</ul>
</div>
<h2 id="create-an-rstudio-project">Create an RStudio project<a class="headerlink" href="#create-an-rstudio-project" title="Permanent link">&para;</a></h2>
<p>One of the first benefits we will take advantage of in RStudio is
something called an <strong>RStudio Project</strong>. An RStudio project allows you
to more easily:</p>
<ul>
<li>Save data, files, variables, packages, etc. related to a specific
     analysis project</li>
<li>Restart work where you left off</li>
<li>Collaborate, especially if you are using version control such as
    <a href="http://swcarpentry.github.io/git-novice/">git</a>.</li>
</ul>
<div class="admonition info">
<ol>
<li>
<p>To create a project, go to the <strong>File</strong> menu, and click <strong>New Project</strong></p>
<p><img alt="image" src="../figures/new_project_window.png" width-_400_="width-&quot;400&quot;" /></p>
</li>
<li>
<p>In the window that opens select <strong>Existing Directory</strong><br />
    Then select <strong>Browse....</strong> <br />
    Choose and then click "~/Documents/IndigiDataIntroToR".  </p>
</li>
<li>
<p>Finally click <code>Create Project</code>. In the
    "Files" tab of your output pane (more about the RStudio layout in a
    moment), you should see an RStudio project file,
    <strong>IndigiDataIntroToR.Rproj</strong>. All RStudio projects end with the
    "<strong>.Rproj</strong>" file extension.</p>
</li>
</ol>
</div>
<h2 id="creating-your-first-r-script">Creating your first R script<a class="headerlink" href="#creating-your-first-r-script" title="Permanent link">&para;</a></h2>
<p>Now that we are ready to start exploring R, we will want to keep a
record of the commands we are using. To do this we can create an R
script:</p>
<p>Click the <code>File</code> menu and select
<code>New File</code> and then <code>R
Script</code>. Before we go any further, save your script by
clicking the save/disk icon that is in the bar above the first line in
the script editor, or click the <code>File</code> menu
and select <code>save</code>. In the "Save File" window
that opens, name your file <strong>"indigidataIntroToR"</strong>. The new script
<strong>indigidataIntroToR.R</strong> should appear under "files" in the output pane.
By convention, R scripts end with the file extension <strong>.R</strong>.</p>
<h2 id="getting-to-work-with-r-navigating-directories">Getting to work with R: navigating directories<a class="headerlink" href="#getting-to-work-with-r-navigating-directories" title="Permanent link">&para;</a></h2>
<p>Now that we have covered the more aesthetic aspects of RStudio, we can
get to work using some commands. We will write, execute, and save the
commands we learn in our <strong>indigidataIntroToR.R</strong> script that is loaded
in the Source pane. First, lets see what directory we are in. To do so,
type the following command into the script:</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<p><code>getwd()</code></p>
</div>
<p>To execute this command, make sure your cursor is on the same line the
command is written. Then click the <code>Run</code>
button that is just above the first line of your script in the header of
the Source pane.</p>
<p>In the console, we expect to see the following output :</p>
<div class="admonition solution">
<p><code>[1] "/home/Documents/IndigiDataIntroToR"</code></p>
</div>
<ul>
<li>Notice, at the Console, you will also see the instruction you executed above the output in blue.</li>
</ul>
<p>Since we will be learning several commands, we may already want to keep
some short notes in our script to explain the purpose of the command.
Entering a <code>#</code> before any line in an R script turns that line into a
comment, which R will not try to interpret as code. Edit your script to
include a comment on the purpose of commands you are learning, e.g.:</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<p>`# this command shows the current working directory </p>
<p>getwd()`</p>
</div>
<div class="admonition question">
<p class="admonition-title">Exercise : Work interactively in R</p>
<p>What happens when you try to enter the <code>getwd()</code> command in the Console pane?</p>
<details class="success">
<summary>Solution</summary>
<p>You will get the same output you did as when you ran <code>getwd()</code> from
 the source. You can run any command in the Console, however,
 executing it from the source script will make it easier for us to
 record what we have done, and ultimately run an entire script,
 instead of entering commands one-by-one. </p>
</details>
</div>
<p>For the purposes of this exercise we want you to be in the directory
<code>"/home/&lt;USERID&gt;/IndigiDataIntroToR"</code>. What if you weren't? You can set your home directory using the <code>setwd()</code> command. Alternatively, RStudio has a helpful button for setting your working directory (<em>note</em> that this is a very simplified way to set your working directory, and this is actually a bit more complex in practice.)</p>
<h2 id="using-functions-in-r-without-needing-to-master-them">Using functions in R, without needing to master them<a class="headerlink" href="#using-functions-in-r-without-needing-to-master-them" title="Permanent link">&para;</a></h2>
<p>A function in R (or any computing language) is a short program that
takes some input and returns some output. Functions may seem like an
advanced topic (and they are), but you have already used at least one
function in R. <code>getwd()</code> is a function! The next sections will help you
understand what is happening in any R script.</p>
<div class="admonition question">
<p class="admonition-title">Exercise: What do these functions do?</p>
<p>Try the following functions by writing them in your script. See if you
can guess what they do, and make sure to add comments to your script
about your assumed purpose. - <code>dir()</code> - <code>sessionInfo()</code> - <code>date()</code> -
<code>Sys.time()</code></p>
<details class="success">
<summary>Solution</summary>
<ul>
<li><code>dir()</code> # Lists files in the working directory</li>
<li><code>sessionInfo()</code> # Gives the version of R and additional info
    including on attached packages</li>
<li><code>date()</code> # Gives the current date</li>
<li><code>Sys.time()</code> # Gives the current time</li>
<li><code>.libPaths()</code> # Shows what libraries are available</li>
</ul>
<p><strong>Notice</strong>: Commands are case sensitive! </p>
</details>
</div>
<p>You have hopefully noticed a pattern - an R function has three key
properties: </p>
<ul>
<li>
<p>Functions have a name (e.g. <code>dir</code>, <code>getwd</code>); note that
functions are case sensitive! </p>
</li>
<li>
<p>Following the name, functions have a
pair of <code>()</code> </p>
</li>
<li>
<p>Inside the parentheses, a function may take 0 or more
arguments.</p>
</li>
</ul>
<p>An argument may be a specific input for your function and/or may modify
the function's behavior. For example the function <code>round()</code> will round a
number with a decimal:</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a># This will round a number to the nearest integer
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>round(3.14)
</code></pre></div>
</div>
<h2 id="getting-help-with-function-arguments">Getting help with function arguments<a class="headerlink" href="#getting-help-with-function-arguments" title="Permanent link">&para;</a></h2>
<p>What if you wanted to round to one significant digit? <code>round()</code> can do
this, but you may first need to read the help to find out how. To see
the help (In R sometimes also called a "vignette") enter a <code>?</code> in front
of the function name:</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<p><code>?round()</code></p>
</div>
<p>The "Help" tab will show you information (often, too much information).
You will slowly learn how to read and make sense of help files. Checking
the "Usage" or "Examples" headings is often a good place to look first.
If you look under "Arguments," we also see what arguments we can pass to
this function to modify its behavior. You can also see a function's
argument using the <code>args()</code> function:</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<p><code>args(round)</code></p>
</div>
<p><code>round()</code> takes two arguments, <code>x</code>, which is the number to be rounded,
and a <code>digits</code> argument. The <code>=</code> sign indicates that a default (in this
case 0) is already set. Since <code>x</code> is not set, <code>round()</code> requires we
provide it, in contrast to <code>digits</code> where R will use the default value 0
unless you explicitly provide a different value. We can explicitly set
the digits parameter when we call the function:</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<p><code>round(3.14159, digits = 2)</code></p>
</div>
<p>Or, R accepts what we call "positional arguments", if you pass a
function arguments separated by commas, R assumes that they are in the
order you saw when we used <code>args()</code>. In the case below that means that
<code>x</code> is 3.14159 and digits is 2.</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<p><code>round(3.14159, 2)</code></p>
</div>
<p>Finally, what if you are using <code>?</code> to get help for a function in a
package not installed on your system, such as when you are running a
script which has dependencies.</p>
<div class="admonition r-project">
<p class="admonition-title">R-project</p>
<p><code>geom_point()</code></p>
</div>
<p>will return an error:</p>
<div class="admonition failure">
<p class="admonition-title">Error</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>Error in .helpForCall(topicExpr, parent.frame()) :
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>   no methods for ‘geom_point’ and no documentation for it as a function
</code></pre></div>
</div>
<p>Use two question marks (i.e. <code>??geom_point()</code>) and R will return results
from a search of the documentation for packages you have installed on
your computer in the "Help" tab. Finally, if you think there should be a
function, for example a statistical test, but you aren't sure what it is
called in R, or what functions may be available, use the <code>help.search()</code>
function.</p>
<div class="admonition question">
<p class="admonition-title">Exercise: Searching for R functions</p>
<p>Use <code>help.search()</code> to find R functions for the following statistical
functions. Remember to put your search query in quotes inside the
function's parentheses.</p>
<ul>
<li>Chi-Squared test</li>
<li>Student t-test</li>
<li>mixed linear model</li>
</ul>
<details class="success solution">
<summary>Success</summary>
<p>While your search results may return several tests, we list a few
you might find: </p>
<ul>
<li>Chi-Squared test: <code>stats::Chisquare</code> </li>
<li>Student t-test: <code>stats::t.test</code> </li>
<li>mixed linear model: <code>stats::lm.glm</code></li>
</ul>
</details>
</div>

              
            </article>
            
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href=".." class="md-footer__link md-footer__link--prev" aria-label="Previous: Home" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Home
            </div>
          </div>
        </a>
      
      
        
        <a href="../2_WorkingWithData/" class="md-footer__link md-footer__link--next" aria-label="Next: Analysing genetic variant data in R" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Analysing genetic variant data in R
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      IndigiDataAotearoa_IntroToR workshop material is licensed under a <a rel="noopener" target="_blank" href="https://www.gnu.org/licenses/gpl-3.0.en.html">GNU General Public License v3.0</a>
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.footer", "content.code.copy", "content.code.annotate"], "search": "../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.6c7ad80a.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>